appgate/secrets.py

```diff
@@ -126,10 +135,78 @@ def isinstance(value: PasswordField) -> bool:
         return isinstance(value, dict) and value.get("type") == "k8s/secret"
 
 
+class AppgateVault:
+    def __init__(self, client: Client):
+        self.client = client
+
+    def authenticate(self):
+        if self.client.is_authenticated():
+            log.info("Successfully authenticated Vault client to %s", self.client.url)
+        else:
+            log.error(
+                "Failed to authenticated Vault client against %s", self.client.url
+            )
+        return self.client
+
+
+class AppgateInClusterVault(AppgateVault):
+    def __init__(self):
+        jwt = open("/var/run/secrets/kubernetes.io/serviceaccount/token").read()
```

Comment:
A bit of a nitpick but better to use a with statement when dealing with files in general, in this case Path.read_Text makes sense I think.
```suggestion
        jwt = Path("/var/run/secrets/kubernetes.io/serviceaccount/token").read_text()
```